### ISSUE TITLE
HTTP/3: Handle platforms that don't support HTTP/3

### ISSUE THREAD

**Issue by JamesNK** [dotnet/aspnetcore#34517]
If HTTP/2 is not supported then it is silently not enabled. We should do the same for HTTP/3 if a HTTP/3 endpoint is configured and requirements are not met. For example: Insufficient Windows version, Linux msquic package not installed.

* If an endpoint is configured with other protocols as well as HTTP/3, e.g. `Http1AndHttp2AndHttp3`, then don't do HTTP/3 and log. Log level INFO? `Http1AndHttp2AndHttp3` will become the default in .NET 7 and we don't want to spam people's logs with WARN on startup.
* If an endpoint is configured with **just** HTTP/3 then error.

Include "See https://aka.ms/aspnet/kestrel/http3reqs for details." in log and exception message.

**Comment by Tratcher** [comment]
Now that the Quic Transport assembly is part of the shared runtime we can start to integrate it into kestrel:
- Add a dependency on the Transport.Quic package
- Call UseQuic automatically, just like we do for sockets.
- Take out the exceptions I added (https://github.com/dotnet/aspnetcore/pull/34434) and add logs similar places to those we have for HTTP/2. The platform detection for quick is a bit different than for http/2, we may want to put the initial check in UseQuic and not register the transport if it's not supported. Then when you try to enable Http3 we log that the transport wasn't available, probably because it's not supported.

**Comment by Tratcher** [comment]
I'm getting started on this now so I've re-assigned it.

**Comment by Tratcher** [comment]
Here's one broken scenario to consider, non-TLS with Http1AndHttp2AndHttp3
```
                        options.ListenAnyIP(5000, listenOptions =>
                        {
                            listenOptions.Protocols = HttpProtocols.Http1AndHttp2AndHttp3;
                        });
```

Throws:
```
Unhandled exception. System.InvalidOperationException: Couldn't find HTTPS configuration for QUIC transport.
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.QuicTransportFactory.BindAsync(EndPoint endpoint, IFeatureCollection features, CancellationToken cancellationToken) in D:\github\AspNetCore\src\Servers\Kestrel\Transport.Quic\src\QuicTransportFactory.cs:line 55
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.TransportManager.BindAsync(EndPoint endPoint, MultiplexedConnectionDelegate multiplexedConnectionDelegate, ListenOptions listenOptions, CancellationToken cancellationToken) in D:\github\AspNetCore\src\Servers\Kestrel\Core\src\Internal\Infrastructure\TransportManager.cs:line 73
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.<>c__DisplayClass29_0`1.<<StartAsync>g__OnBind|0>d.MoveNext() in D:\github\AspNetCore\src\Servers\Kestrel\Core\src\Internal\KestrelServerImpl.cs:line 196
--- End of stack trace from previous location ---
```

When doing non-TLS with Http1AndHttp2 we silently default to Http1. We should do the same in this case.

**Comment by Tratcher** [comment]
Fixed by #34877
